src/fluree/db/full_text.clj

```diff
@@ -21,47 +23,66 @@
            org.apache.lucene.analysis.bn.BengaliAnalyzer
            org.apache.lucene.analysis.br.BrazilianAnalyzer
            org.apache.lucene.analysis.fr.FrenchAnalyzer
-           org.apache.lucene.index.IndexWriter))
-
-(defprotocol BlockRegistry
-  (read-current-block [r])
-  (register-block [r status])
-  (clear [r]))
-
-(defrecord DiskRegistry [^File file]
-  BlockRegistry
-  (read-current-block
-    [_]
-    (when (.exists file)
-      (-> file slurp edn/read-string)))
-  (register-block
-    [_ status]
-    (->> status prn-str (spit file)))
-  (clear
-    [_]
-    (when (.exists file)
-      (io/delete-file file))))
-
-(defn disk-registry
+           org.apache.lucene.index.IndexWriter
+           org.apache.lucene.index.IndexReader
+           org.apache.lucene.index.IndexNotFoundException
+           org.apache.lucene.store.Directory))
+
+(def search-limit Integer/MAX_VALUE)
+
+(defrecord Index [^Directory storage ^Analyzer analyzer block-registry]
+  Closeable
+  (close [_]
+    (.close analyzer)
+    (.close storage)))
+
+;; TODO: determine size impact of these analyzers - can we package them
```

Comment:
Do we have a Jira ticket for this? It would be good to analyze before we release since if we decide to split them out later that would be a breaking change. I'd be happy to help!

Reply:
Ah, nvm, I see this has been around for a bit.

Reply:
This TODO predates me so I'm not sure if we have a ticket for it or not, but I think we _should_ have a ticket if we don't already. I'll poke around and add it if I can't find it.